### ISSUE TITLE
No looping of subtitles for live streams

### ISSUE THREAD

**Issue by Romantic-LiXuefeng** [google/shaka-streamer#34]
#32 
> input_looped_file_config.yaml

```
# List of inputs.
inputs:
    # The type of input.
  - input_type: looped_file
    # Name of the input file.
    # This example can be downloaded from https://storage.googleapis.com/shaka-streamer-assets/sample-inputs/Sintel.2010.720p.mkv
    name: Sintel.2010.720p.mkv
    # The track number.
    track_num: 0
    # The media type of the input. Can be audio or video.
    media_type: video
    # Frame rate in seconds.
    frame_rate: 24.0
    # Resolution of the input.
    resolution: 720p

    # The type of input.
  - input_type: looped_file
    # A second track (audio) from the same input file.
    name: Sintel.2010.720p.mkv
    track_num: 1
    media_type: audio

    # The type of input.
  - input_type: looped_file
    # https://storage.googleapis.com/shaka-streamer-assets/sample-inputs/Sintel.2010.Chinese.vtt
    name: Sintel.2010.Chinese.vtt
    media_type: text
    track_num: 2
    language: zh
```

> pipeline_live_config.yaml

```
# Streaming mode.  Can be live or vod.
streaming_mode: live

# A list of resolutions to encode.
resolutions:
  - 720p
  - 480p

# The number of audio channels to output.
channels: 2

# The codecs to encode with.
audio_codecs:
  - aac
video_codecs:
  - h264

# Manifest format (dash, hls, or both)
manifest_format:
  # - dash
  - hls

# Length of each segment in seconds.
segment_size: 4

# Availability window, or the number of seconds a segment remains available.
availability_window: 300

# Presentation delay, or how far back from the edge the player should be.
presentation_delay: 30

# Update period, or how often the player should fetch a new manifest.
update_period: 8
```

**Comment by joeyparrish** [comment]
I found a couple of issues with this in the latest code.

First, we just introduced auto-detection of several input features and changed the definition of track_num in an incompatible way.  To run this on the latest version, you'll need to update your input config to drop the "track_num" field from all inputs.  You can also simplify and drop "frame_rate", and "resolution" fields from the video input.

Second, this triggers a bug that we introduced with the custom bitrates feature.  I'll have a fix for that soon.

**Comment by joeyparrish** [comment]
Finally, with those issues fixed, the remaining issue is that we don't have a way to loop subtitles at the moment.  Subtitle inputs bypass the transcoder and go straight to the packager.

I just checked to make sure, and I can confirm that ffmpeg doesn't support looping subtitle inputs.

The subtitles are being packaged into the output, though.

**Comment by joeyparrish** [comment]
We will add a restriction so that `media_type: text` combined with `input_type: looped_file` will be rejected with a clear error message.  We will also update the docs to reflect this.  The changes will be released in v0.3.0.

**Comment by joeyparrish** [comment]
Changes have been made to clarify the docs and reject these unsupported combinations of options.  Please let us know if you have any further issues.  Thanks!
